src/Microsoft.AspNet.OData/EnableQueryAttribute.cs

```diff
@@ -71,6 +73,37 @@ public override void OnActionExecuted(HttpActionExecutedContext actionExecutedCo
                 throw Error.Argument("actionExecutedContext", SRResources.ActionExecutedContextMustHaveActionContext);
             }
 
+            bool hasExpandQuery = false;
+            string queryString = request.RequestUri.Query;
+            var absoluteUri = request.RequestUri.AbsoluteUri;
+            bool hasEmptyQueryString = string.IsNullOrEmpty(queryString);
+            string baseUriWithPath = hasEmptyQueryString ? absoluteUri : absoluteUri.Substring(0, absoluteUri.IndexOf('?', 0));
+
+            if (!hasEmptyQueryString)
+            {
+                var queries = queryString.Split('&');
+
+                hasExpandQuery = queries.Any(x =>
+                {
+                    var xToUpperInvariant = x.ToUpperInvariant();
+
+                    return xToUpperInvariant.StartsWith("?$EXPAND", StringComparison.OrdinalIgnoreCase) || xToUpperInvariant.StartsWith("$EXPAND", StringComparison.OrdinalIgnoreCase) || xToUpperInvariant.StartsWith("EXPAND", StringComparison.OrdinalIgnoreCase);
+                });
+            }
+
+            // Create a $expand query string if 1) It's a POST request 2) if there is no expand query string.
+            if (String.Equals(actionExecutedContext.Request.Method.ToString(), "post", StringComparison.OrdinalIgnoreCase) && !hasExpandQuery)
```

Comment:
What's the type of `Request.Method`. Just curious if it can be compared without performing a string conversion (e.g. if it's an enum).

But this is not a major concern.

Reply:
Did a comparison using [HttpMethod.Post](https://learn.microsoft.com/en-us/previous-versions/visualstudio/hh193683(v=vs.118))